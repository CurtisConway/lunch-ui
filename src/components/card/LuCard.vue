<template>
  <LuColumn :tag="tag" :class="defaultClassList">
    <slot></slot>
  </LuColumn>
</template>

<script>
import spacing from '../../assets/js/mixins/spacing';
import elevation from '../../assets/js/mixins/elevation';
import borders from '../../assets/js/mixins/borders';
import LuColumn from '../grid/LuColumn.vue';
import colors from '../../assets/js/mixins/colors';

export default {
  name: 'LuCard',
  mixins: [spacing, elevation, borders, colors],
  components: {
    LuColumn,
  },
  props: {
    tag: {
      type: String,
      default: () => 'div',
    },
    elevation: {
      type: [String, Number],
      default: () => 4,
    },
  },
  computed: {
    defaultClassList() {
      return [
        'lu-card',
        this.paddingClasses,
        this.marginClasses,
        this.elevationClass,
        this.borderClasses,
        this.bgColorClass,
      ];
    },
  },
};
</script>
