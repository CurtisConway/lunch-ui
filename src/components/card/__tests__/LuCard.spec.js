import Vue from 'vue';
import Router from 'vue-router';

import LuCard from '../LuCard.vue';

import {
  createLocalVue,
  mount,
} from '@vue/test-utils';
import { compileToFunctions } from 'vue-template-compiler'

describe('LuCard.vue', () => {
  let mountFunction;
  let router = Router;
  let localVue = typeof Vue;

  beforeEach(() => {
    router = new Router();
    localVue = createLocalVue();
    localVue.use(Router);

    mountFunction = (options = {}) => {
      return mount(LuCard, {
        localVue,
        router,
        ...options,
      });
    }
  })

  it('should render component and match snapshot', () => {
    expect(mountFunction().html()).toMatchSnapshot();
  });
});
