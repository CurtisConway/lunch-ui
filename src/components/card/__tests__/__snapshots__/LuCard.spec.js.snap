// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LuCard.vue should render component and match snapshot 1`] = `<div class="lu-card"></div>`;
